flask_appbuilder/menu.py

```diff
@@ -123,3 +151,54 @@ def add_separator(self, category=""):
             raise Exception(
                 "Menu separator does not have correct category {}".format(category)
             )
+
+
+class MenuApi(BaseApi):
+    resource_name = "menu"
+
+    @expose('/', methods=["GET"])
+    def get_menu_data(self):
+        """An endpoint for retreiving the menu.
+        ---
+        get:
+          responses:
+            200:
+              description: Get menu data
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      result:
+                        type: array
+                        items:
+                          oneOf:
+                            Seperator:
+                              type: string
+                              enum:
+                              - '-'
+                            MenuObject:
+                              type: object
+                              properties:
+                                name:
+                                  type: string
+                                label:
+                                  type: string
+                                icon:
+                                  type: string
+                                url:
+                                  type: string
+                                childs:
+                                  $ref: \
+        '#/paths/menu/data/get/reponses/200/content/application/json/schema/properties/result'
+                              required:
+                                - name
+                                - label
+                                - icon
+        """
+        return self.response(200, result=current_app.appbuilder.menu.get_data())
+
+
+class MenuApiManager(BaseManager):
+    def register_views(self):
+        self.appbuilder.add_api(MenuApi)
```

Comment:
Let's create a new option key for this, take a look at: https://github.com/dpgaspar/Flask-AppBuilder/blob/master/flask_appbuilder/security/manager.py#L581

Reply:
You mean expose a config to enable/disable this endpoint? something like `FAB_EXPOSE_MENU_API`?

Reply:
yep